packages/core/src/data-access/documentVersions.ts

```diff
@@ -64,44 +82,39 @@ export async function getDocumentsAtCommit(
   const documentsAtPreviousMergedCommits =
     documentsAtPreviousMergedCommitsResult.map((d) => d.document_versions)
 
-  if (maxMergedAt) {
+  if (maxMergedAt !== null) {
     // Referenced commit is merged. No additional documents to return.
     return Result.ok(documentsAtPreviousMergedCommits)
   }
 
-  const commitResult = await findCommit({ projectId, uuid: commitUuid })
-  if (commitResult.error) return commitResult
-
-  const commit = commitResult.unwrap()
-
   const documentsAtDraftResult = await tx
     .select()
     .from(documentVersions)
     .innerJoin(commits, eq(commits.id, documentVersions.commitId))
-    .where(eq(commits.id, commit.id))
+    .where(eq(commits.id, commitId))
 
   const documentsAtDraft = documentsAtDraftResult.map(
     (d) => d.document_versions,
   )
   const totalDocuments = documentsAtPreviousMergedCommits
-    .filter((d) =>
-      documentsAtDraft.find((d2) => d2.documentUuid !== d.documentUuid),
+    .filter(
+      (d) =>
+        documentsAtDraft.find((d2) => d2.documentUuid === d.documentUuid) ===
+        undefined,
     )
     .concat(documentsAtDraft)
 
   return Result.ok(totalDocuments)
 }
```

Comment:
hmm hadn't seen this method before. It's a clever query! You should be able to improve code by splitting in smaller chunks. You really only need three operations for getting documents at a particular commit afaict:

- fetch main branch documents
- fetch commit documents
- merge main branch and draft documents

Can we split this method in these 3 concepts?